File: restaurants/seaside/get_menu.py
from urllib import request
from bs4 import BeautifulSoup
import json
import datetime

from os import path
directory = path.dirname(path.realpath(__file__))


def main():
    html = request.urlopen("http://seaside.kvartersmenyn.se/").read().decode()

    html_tree = BeautifulSoup(html, features="html.parser")

    relevant_html = html_tree.find_all(class_="meny")[0].contents

    now = datetime.datetime.now().isocalendar()

    menu = {
        "year": now[0],
        "week": now[1],
        "days": {}
    }

    for day in range(0, 5):
        i = day * 11 + 2

        dishes = []
        
        for dish_index in range(4):
            try:
                dish = relevant_html[i + dish_index * 2].strip()
                dish_array = dish.split()
                
                dishes.append({
                    "title": dish_array[0],
                    "description": " ".join(dish_array[1:])
                })
            except IndexError:
                pass
        
        if len(dishes) == 0:
            dishes = ["N/A"]

        menu["days"][day] = dishes
    
    fp = open(directory + "/menu.json", "w", encoding="utf-8")

    json.dump(menu, fp, ensure_ascii=False)


main()

File: restaurants/seaside/getMenu.js
const cheerio = require("cheerio")
const http = require("http")

function getHtml() {
    let options = {
        hostname: 'seaside.kvartersmenyn.se',
        port: 80,
        method: "GET"
    }
    
    return new Promise((resolve, reject) => {
        let req = http.request(options, (res) => {
            let data = ""
            res.on("data", d => data += d)
            res.on("end", () => {
                resolve(data)
            })
        })
        
        req.end()
    })
}

module.exports = async () => {
    let html = await getHtml()
    const $ = cheerio.load(html)


    let relevantHtml = $(".meny")[0].children

    let now = new Date()

    let menu = {
        restaurant: "Restaurang Seaside",
        year: now.getFullYear(),
        week: getWeek(now),
        days: {}
    }

    for (let day = 0; day < 5; day++) {
        let i = day * 11 + 2
        let dishes = []
        
        for (let dishIndex = 0; dishIndex < 4; dishIndex++) {
            try {
                let dish = relevantHtml[i + dishIndex * 2].data.trim()
                if (dish && dish !== "") {
                    let dishArray = dish.split(" ")

                    dishes.push({
                        title: dishArray[0],
                        description: dishArray.slice(1).join(" ")
                    })
                }
            } catch (e){
                // Do nothing
            }
        }

        menu.days[day] = dishes
    }
    return menu
}

function getWeek(date) {
    var year = date.getFullYear();
    var oneJan = new Date("Jan 1 " + year);
    var oneJanDay = (oneJan.getDay() + 6) % 6 
    var day = Math.floor((date.getTime() - oneJan.getTime()) / 86400000);
    var week = Math.floor(day / 7);

    if (oneJanDay < 4) {
        week++;
    }
    return week;
}
